### DISCUSSION TITLE
Adding Thumb for Rectangle Control

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#7673] [Q&A] [answered]
Hi Guys,

Have a question about creating a rectangle and want to add four thumbs to doing drag and drop / Resize functions.
I wanted to create a adorner from this custom control.

For the moment, I'm using AdornerLayer to add this Adorner as a Rectangle but I'm struggling to add thumbs.

Have you some clues about it ?

**Comment by maxkatz6** [comment]
Hi! Very late answer, but probably I can answer for the future, if somebody will look for it.

Big sample project what can be used as a reference: https://github.com/wieslawsoltes/NodeEditor (doesn't use adorner layer).
But simpler clue: you can add ellipses on the corners and set AdornerLayer.IsClipEnabled attached property to false.

**Comment by wieslawsoltes** [comment, marked as answer]
I have some experimental example

https://github.com/wieslawsoltes/PerspectiveDemo/tree/Resizing/PerspectiveDemo

https://github.com/wieslawsoltes/PerspectiveDemo/blob/b2ea71f6c800933393a2443ee445e46e3d8faf9e/PerspectiveDemo/SelectedAdorner.axaml#L9

https://github.com/wieslawsoltes/PerspectiveDemo/blob/b2ea71f6c800933393a2443ee445e46e3d8faf9e/PerspectiveDemo/SelectedAdorner.axaml.cs#L9

**Comment by Tenjim** [comment]
Thank guys, I'll look on it !
